### PR TITLE
doc: fix pdf export

### PR DESCRIPTION
**Pull request by roipoussiere**
Documentation export in pdf was not working. This PR fix this by:

- removing the sphinx build option `-W` (a.k.a `--fail-on-warning`) in conf.py in order to continue the build even if the documentation contains warnings (see #11972) __*__;
- using Xelatex engine in order to support the unicode characters included in the doc;
- configuring the builder to use png images as fallback, since gif images are not supported by the LaTeX engine;
- adding png fallback images for each gif images.

__*__: I also removed the unnecessary `--keep-going`, these two options are opposite.

### CODE REVIEW

**Comment by gasman** [comment]
Thanks @roipoussiere! Just to highlight that the build errors have been addressed in #12044 which is currently awaiting review, so if that gets approved we won't need to remove the `-W` flag.

File: docs/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -W --keep-going -n -jauto
```
Comment (laymonage) [inline docs/Makefile:5]: I'm pretty sure we want to keep this in the future, so it fails loudly when we have legitimate errors. As you pointed out, we have #11972 to track this, and there's an open PR to fix it in #12044. That being said, I need to review that PR soon before I can review yours, so we can fix both problems. Will do when I find the time... 🙂

**Review by laymonage — COMMENTED** [review]
Thanks for the PR! I haven't fully reviewed it but I just wanted to flag how this relates to #11972, just in case you're wondering if this PR does not get reviewed soon.

**Comment by roipoussiere** [comment]
Thanks for the review. I added the `--fail-on-warning` sphinx option in the commit.

File: docs/Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+SPHINXOPTS    = --fail-on-warning -n -jauto
```
Comment (gasman) [inline docs/Makefile:5]: This change is causing the HTML build to fail on my machine:

```sh
$ make html
sphinx-build -b html -d _build/doctrees  --fail-on-warning -n -jauto . _build/html
usage: sphinx-build [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
sphinx-build: error: unrecognized arguments: --fail-on-warning
make: *** [html] Error 2
```

**Review by gasman — CHANGES_REQUESTED** [review]

File: docs/Makefile (hunk 2 above)
Comment (gasman) [reply docs/Makefile:5]: Looks like the long option was added in Sphinx 7.3, so this just needs a version bump in setup.py.

**Comment by gasman** [comment]
Merged in 5a8c33f34a36a038ab8cdaedddf19fe4d5750cf5 (with Sphinx version bump to >=7.3 added).

@roipoussiere - I've added you to the contributors list as "Nathanaël" - let me know if you'd like this changed to a full name.

**Comment by roipoussiere** [comment]
Thanks! If it's not too late you can add my last name as well (Nathanaël Jourdane) for consistency with other contributors in the list.

**Comment by gasman** [comment]
No problem! Updated in efc419a3a1 (main) and fe9a2c493f8479e82adc80797137b723f2ce95a3 (stable/6.2.x).

**Pull request closed without merging**